hbase-protocol-shaded/src/main/protobuf/server/master/Master.proto

```diff
@@ -693,6 +694,14 @@ message SwitchExceedThrottleQuotaResponse {
   required bool previous_exceed_throttle_quota_enabled = 1;
 }
 
+message BalancerDecisionsRequest {
```

Comment:
Remove. Use getLogEntries() instead.

Reply:
Using common API for proto seems almost impossible. One use-case of slowLog is being handled by RSRpc services whereas the other for balancerDecision is being managed by MasterRpc services. Both have different sets of Rpc implementations written and handled at different places.
Moreover, at RPC level, we can't achieve abstraction since protobuf doesn't support `is-a` relationship among proto `messages`.

Reply:
Not impossible at all.

getRingEntries RPC in RS RPC servcies

getRingEntries RPC in master RPC services

> we can't achieve abstraction since protobuf doesn't support is-a relationship among proto messages.

You have an outer message type that is a class name and a byte string, and the inner protobuf is stuffed into the byte string while the class name is set to the type you need to use to deserialize.

Reply:
> You have an outer message type that is a class name and a byte string, and the protobuf for the inner type is serialized into the byte string while the class name field of the outer type is set to the type to use for parsing the inner type.

Thanks a lot, I was not aware of this usage. Yes, passing an object as ByteString should be good enough.